contrib/new_post.py

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/env python3
+"""Create a new post."""
+import argparse
+import datetime
+import json
+import os
+import sys
+import typing
+import urllib.request
+
+# These correspond to the GitHub labels used by Bitcoin Core.
+DESIRED_COMPONENTS = [
+  'Block storage',
+  'Build system',
+  'Consensus',
+  'Data corruption',
+  'Descriptors',
+  'Docs',
+  'GUI',
+  'Interfaces',
+  'Mempool',
+  'Mining',
+  'P2P',
+  'Privacy',
+  'PSBT',
+  'Refactoring',
+  'Resource usage',
+  'RPC/REST/ZMQ',
+  'Scripts and tools',
+  'Settings',
+  'Tests',
+  'TX fees and policy',
+  'Utils/log/libs',
+  'UTXO Db and Indexes',
+  'Validation',
+  'Wallet',
+]
+COMPONENTS = list(map(lambda x: x.lower(), DESIRED_COMPONENTS))
+
+# Some PRs contain undesired words (here, single characters) immediately after
+# the prefix. Add the characters or words here to strip them from the title.
+UNDESIRED_PR_TITLE_WORDS = ['-', '_']
+
+GITHUB_API_URL = 'https://api.github.com/repos/bitcoin/bitcoin/pulls'
+
+class PullRequest(typing.NamedTuple):
+    number: int
+    title: str
+    labels: typing.List[str]
+    user: str
+
+def validate_date(date_in):
+    date = datetime.date.fromisoformat(date_in)
+    return date.isoformat()
+
+def next_wednesday():
+    today = datetime.date.today()
+    days_ahead = 2 - today.weekday()
+    if days_ahead <= 0:  # Target day already happened this week
+        days_ahead += 7
+    return (today + datetime.timedelta(days_ahead)).isoformat()
+
+def is_a_component(prefix):
+    # Boolean indicating whether `prefix` (without any final "s") is a component.
+    # Iterates through the COMPONENTS array and exits with true at the first
+    # instance where `prefix` is a substring of the component; otherwise false.
+    return any([prefix in component for component in COMPONENTS])
+
+def parse_title(title):
+    words = title.split()
+
+    # Remove unwanted characters from the title
+    if words[1] in UNDESIRED_PR_TITLE_WORDS:
+        words.pop(1)
+
+    # If the first word is a component with square brackets or colon, drop first word.
+    words[0] = words[0].lower()  # lowercase the first word in place
+    prefix = words[0].replace(':[]', '')  # prefix is first word stripped of :[] chars
```

Comment:
This only replaces strings that exactly match ":[]", so it doesn't work here. Could chain `.replace()` or add a helper function, but tbh I think this logic can be cleared up a bit anyway, what do you think about this? Much more straightforward to understand, imo.

<details>
<summary>git diff</summary>

```diff
diff --git a/contrib/new_post.py b/contrib/new_post.py
index 051790f..3ccfeae 100755
--- a/contrib/new_post.py
+++ b/contrib/new_post.py
@@ -69,15 +69,15 @@ def is_a_component(prefix):
 def parse_title(title):
     words = title.split()
 
-    # Remove unwanted characters from the title
-    if words[1] in UNDESIRED_PR_TITLE_WORDS:
-        words.pop(1)
-
-    # If the first word is a component with square brackets or colon, drop first word.
-    words[0] = words[0].lower()  # lowercase the first word in place
-    prefix = words[0].replace(':[]', '')  # prefix is first word stripped of :[] chars
-    if words[0] != prefix and is_a_component(prefix):
-        words.pop(0)
+    # Remove prefixes and unwanted characters from the beginning of the title
+    for word in list(words):
+        if (word in UNDESIRED_PR_TITLE_WORDS
+                or word.endswith(':')  # usually a component prefix
+                or (word.startswith('[') and word.endswith(']'))):  # usually a component prefix
+            words.pop(0)
+        else:
+            # Continue as soon as we've found an allowed word
+            break
 
     # Capitalize the first word
     words[0] = words[0].capitalize()

```
</details>

Reply:
Yep, agree that's much nicer.